File: app/src/main/java/com/yuriy/openradio/gabor/view/list/BothSideCoordinatorLayout.java
package com.yuriy.openradio.gabor.view.list;

import android.annotation.TargetApi;
import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.view.View;

import com.xenione.libs.swipemaker.AbsCoordinatorLayout;
import com.xenione.libs.swipemaker.SwipeLayout;
import com.yuriy.openradio.gabor.R;

/**
 * Created on 06/04/16.
 */
public class BothSideCoordinatorLayout extends AbsCoordinatorLayout {

    public BothSideCoordinatorLayout(final Context context) {
        super(context);
    }

    public BothSideCoordinatorLayout(final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    public BothSideCoordinatorLayout(final Context context, final AttributeSet attrs, final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public BothSideCoordinatorLayout(final Context context, final AttributeSet attrs,
                                     final int defStyleAttr, final int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    public void doInitialViewsLocation() {
        final SwipeLayout foregroundView = findViewById(R.id.foreground_view);
        final View settings = findViewById(R.id.settings_btn_view);
        final View favorite = findViewById(R.id.favorite_view);
        foregroundView.anchor(-favorite.getWidth(), 0, settings.getRight());
    }

    @Override
    public void onTranslateChange(float global, int index, float relative) {
    }
}

File: app/src/main/java/com/yuriy/openradio/mobile/view/list/BothSideCoordinatorLayout.kt
package com.yuriy.openradio.mobile.view.list

import android.content.Context
import android.util.AttributeSet
import com.xenione.libs.swipemaker.AbsCoordinatorLayout
import com.xenione.libs.swipemaker.SwipeLayout
import com.yuriy.openradio.mobile.R
import com.yuriy.openradio.shared.utils.findView

/**
 * Created on 06/04/16.
 */
class BothSideCoordinatorLayout : AbsCoordinatorLayout {

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun doInitialViewsLocation() {
        val foregroundView = findViewById<SwipeLayout>(R.id.foreground_view)
        val settings = findView(R.id.settings_btn_view)
        val favorite = findView(R.id.favorite_view)
        foregroundView.anchor(-favorite.width, 0, settings.right)
    }

    override fun onTranslateChange(globalPercent: Float, index: Int, relativePercent: Float) {}
}
